lib/plugin.js

```diff
@@ -1,16 +1,33 @@
 import { GraphQLClient } from 'graphql-request';
+import merge from 'lodash/merge';
 
 if (<%= options.useFetchPolyfill %>) {
   require('cross-fetch/polyfill');
 }
 
 export default (ctx, inject) => {
-  const clientOptions = <%= serialize(options.options) %>;
-  const contextEndpoint = (ctx.$config || {}).GRAPHQL_ENDPOINT;
+  const baseOptions = <%= serialize(options) %>;
+  const clients = <%= serialize(Object.keys(options.clients)) %>;
 
-  const client = new GraphQLClient(contextEndpoint || '<%= options.endpoint || '' %>', clientOptions);
+  const graphClients = Object.entries(baseOptions.clients)
+    .map(([clientName, config]) => {
+      const runtimeConfig = (
+        ctx.$config
+        && ctx.$config.graphql
+        && ctx.$config.graphql.clients
+        && ctx.$config.graphql.clients[clientName]
+      ) || {};
 
-  // Inject the graphql client to the context as $graphql
-  ctx.$graphql = client;
-  inject('graphql', client);
+      const clientConfig = merge({
+        options: baseOptions.options,
+      }, config, runtimeConfig);
+
+      return [clientName, new GraphQLClient(clientConfig.endpoint, clientConfig.options)];
+    }).reduce((clients, [name, client]) => ({
```

Comment:
Maybe we should just use `.reduce` instead of `.map` + `.reduce` ?

```js
const graphClients = Object.entries(baseOptions.clients).reduce(
    (clients, [clientName, config]) => {
      const runtimeConfig =
        (ctx.$config &&
          ctx.$config.graphql &&
          ctx.$config.graphql.clients &&
          ctx.$config.graphql.clients[clientName]) ||
        {};

      const clientConfig = Object.assign(
        {},
        {
          ...{ options: baseOptions.options },
          ...config,
          ...runtimeConfig,
        }
      );

      return {
        ...clients,
        [clientName]: new GraphQLClient(clientConfig.endpoint, clientConfig.options),
      };
    },
    {}
  );
```